libminifi/test/TestBase.h

```diff
@@ -427,12 +427,18 @@ class TestController {
     flow_version_ = std::make_shared<minifi::state::response::FlowVersion>("test", "test", "test");
   }
 
+  std::shared_ptr<TestPlan> createPlan(std::shared_ptr<minifi::Configure> configuration = nullptr, const char* state_dir = nullptr) {
+    return createPlan<core::repository::VolatileContentRepository>(configuration, state_dir);
+  }
+
+  template<class ContentRepositoryClass>
   std::shared_ptr<TestPlan> createPlan(std::shared_ptr<minifi::Configure> configuration = nullptr, const char* state_dir = nullptr) {
     if (configuration == nullptr) {
       configuration = std::make_shared<minifi::Configure>();
       configuration->set(minifi::Configure::nifi_state_management_provider_local_class_name, "UnorderedMapKeyValueStoreService");
+      configuration->set(minifi::Configure::nifi_dbcontent_repository_directory_default, createTempDirectory());
     }
-    std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::VolatileContentRepository>();
+    std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<ContentRepositoryClass>();
```

Comment:
Instead of making the whole function a template, I think it would be simpler to make the content repository an optional parameter.

Reply:
Good idea changed it in https://github.com/apache/nifi-minifi-cpp/pull/1177/commits/3026364fda3a6b5b6512a3ba2c3b0a6dccfd7501